### PR TITLE
Feature/hardening

### PR DESCRIPTION
**Pull request by henryju**

### CODE REVIEW

File: org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/tracking/ServerIssueTrackable.java [hunk 1]
```diff
@@ -99,6 +99,22 @@ public String getSeverity() {
   }
 
   @Override
+  public String getRawSeverity() {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public String getType() {
+    // FIXME We are not able to get server side issue type
```
Comment (SonarTech) [inline org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/tracking/ServerIssueTrackable.java]: MAJOR Take the required action to fix the issue indicated by this comment. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1134)

File: org.sonarlint.eclipse.ui/src/org/sonarlint/eclipse/ui/internal/markers/ShowIssueFlowsMarkerResolver.java [hunk 2]
```diff
@@ -125,7 +125,7 @@ public static void showAnnotations(IMarker marker, ITextEditor textEditor) {
     try {
       positions = doc.getPositions(MarkerUtils.SONARLINT_EXTRA_POSITIONS_CATEGORY);
     } catch (BadPositionCategoryException e) {
```
Comment (SonarTech) [inline org.sonarlint.eclipse.ui/src/org/sonarlint/eclipse/ui/internal/markers/ShowIssueFlowsMarkerResolver.java]: MAJOR Either log or rethrow this exception. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1166)

File: org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/tracking/CombinedTrackable.java [hunk 3]
```diff
@@ -37,6 +39,10 @@ public CombinedTrackable(Trackable serverIssue, Trackable currentIssue) {
     this.creationDate = serverIssue.getCreationDate();
     this.resolved = serverIssue.isResolved();
     this.assignee = serverIssue.getAssignee();
+    this.severity = serverIssue.getSeverity();
+    // TODO when we are able to get server side issue type
```
Comment (SonarTech) [inline org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/tracking/CombinedTrackable.java]: INFO Complete the task associated to this TODO comment. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1135)

File: org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/tracking/CombinedTrackable.java [hunk 4]
```diff
@@ -37,6 +39,10 @@ public CombinedTrackable(Trackable serverIssue, Trackable currentIssue) {
...
+    // this.type = serverIssue.getType();
```
Comment (SonarTech) [inline org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/tracking/CombinedTrackable.java]: MAJOR This block of commented-out lines of code should be removed. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACommentedOutCodeLine)

File: org.sonarlint.eclipse.ui/src/org/sonarlint/eclipse/ui/internal/views/locations/IssueLocationsView.java [hunk 5]
```diff
@@ -235,10 +235,8 @@ public IMarker getMarker() {
 
             p -> (ExtraPosition) p)
           .filter(filter).collect(Collectors.toList());
-      } catch (
-
-      BadPositionCategoryException e) {
-        SonarLintLogger.get().error("Unable to read positions", e);
+      } catch (BadPositionCategoryException e) {
```
Comment (SonarTech) [inline org.sonarlint.eclipse.ui/src/org/sonarlint/eclipse/ui/internal/views/locations/IssueLocationsView.java]: MAJOR Either log or rethrow this exception. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1166)

**Comment by SonarTech** [comment]
SonarQube analysis reported 5 issues
* ![MAJOR][MAJOR] 4 major
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

[INFO]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO'
[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

File: org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/tracking/Trackable.java [hunk 6]
```diff
@@ -62,8 +62,26 @@
...
+
+  /**
+   * Can be overriden by server side issue in connected mode
+   */
+  String getType();
+
+  /**
+   * Original type reported by the analyzer
+   */
+  String getRawType();
```
Comment (janos-ss) [inline org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/tracking/Trackable.java]: I think "raw" usually means "unprocessed", not ready for use. I would have used `getOriginalType` and `getOriginalSeverity`, but it's up to you.

**Pull request merged**